src/lib/operators/import.cpp

```diff
@@ -0,0 +1,82 @@
+#include "import.hpp"
+
+#include <boost/algorithm/string.hpp>
+
+#include "hyrise.hpp"
+#include "import_export/binary/binary_parser.hpp"
+#include "import_export/csv/csv_parser.hpp"
+#include "utils/assert.hpp"
+#include "utils/load_table.hpp"
+
+namespace opossum {
+
+Import::Import(const std::string& filename, const std::optional<std::string>& tablename, const ChunkOffset chunk_size,
+               const FileType type, const std::optional<CsvMeta>& csv_meta)
+    : AbstractReadOnlyOperator(OperatorType::Import),
+      _filename(filename),
+      _tablename(tablename),
+      _chunk_size(chunk_size),
+      _type(type),
+      _csv_meta(csv_meta) {}
+
+const std::string& Import::name() const {
+  static const auto name = std::string{"Import"};
+  return name;
+}
+
+std::shared_ptr<const Table> Import::_on_execute() {
+  if (_tablename && Hyrise::get().storage_manager.has_table(*_tablename)) {
+    return Hyrise::get().storage_manager.get_table(*_tablename);
+  }

```

Comment:
Is this really what a user would expect?

Reply:
It was surprising for me, but I kept this since I thought this was intended. I think as we now can get a table directly by calling a parser, we could make the table name non-optional and always add it to the storage manager if there is no real reason for not doing so.

Reply:
I like not having to type more than I have to and for the table name being optional. My question was more about: "Would a user be surprised if they submit an import statement to be given an old, potentially completely unrelated table?"

Reply:
At least for me, this was surprising, but I thought this was intended. Anyways, if we do not return the table anymore to be in line with column_expressions, I would just add the table to the storage manager. The problem then would be that actual nothing will happen when no tablename is provided. As a tablename is always given from Hyrise interfaces (import statement needs a tablename, console uses filename if necessary), making it mandatory would not change anything there.